src/elements/_has-arrow.scss

```diff
@@ -0,0 +1,279 @@
+
+@mixin __children { // was #Arrow.children({})
+  &::before,
+  &::after {
+    @content;
+  }
+}
+
+@mixin __shadow($shadowSide: none) { // was #Arrow.shadow({})
+  $borderShadow: 0 0 0 $border-width var(--hxBorderColor, $border-color);
+  $leftShadow: -3px 3px 3px 0 $shadow-color;
+  $rightShadow: 3px 3px 3px 0 $shadow-color;
+
+  &::before {
+    @if $shadowSide == left {
+      box-shadow: $borderShadow, $leftShadow;
+    } @else if $shadowSide == right {
+      box-shadow: $borderShadow, $rightShadow;
+    } @else {
+      box-shadow: $borderShadow;
+    }
+
+    @content;
+  }
+}
+
+@mixin __body { // was #Arrow.body({ ... })
+  &::after {
+    @content;
+  }
+}
+
+// NOTE: not an exact 1:1 replacement for #Arrow.rotate(deg, from)
+@mixin __rotateChildren($deg: 0deg, $from: center) {
+  @include __children {
+    transform-origin: $from;
+    transform: rotate($deg);
+    @content;
+  }
+}
+
+.has-arrow {
+  margin: 0;
+  position: relative;
+
+  @include __shadow {
+    z-index: -1;
+  }
+
+  @include __children {
+    content: '';
+    display: block;
+    height: $arrow-box-side;
+    position: absolute;
+    width: $arrow-box-side;
+  }
+
+  /// ======================================== ///
+  ///  TOP POSITIONS
+  /// ======================================== ///
+  &[position^="top"] {
+    margin-bottom: $arrow-margin;
+
+    @include __children {
+      bottom: -($arrow-margin);
+    }
+
+    @include __body {
+      background-image: linear-gradient(
+        to bottom left,
+        transparent 50%,
+        var(--hxBackgroundColor, $gray-0) 50%
+      );
+    }
+  }
+
+  &[position="top"], // DEPRECATED: remove in v1.0.0
+  &[position="top-center"] {
+    @include __shadow(left);
+    @include __rotateChildren(-45deg, bottom left) {
+      left: 50%;
+    }
+  }
+
+  &[position="top-right"] {
+    @include __shadow(left);
+    @include __rotateChildren(-45deg, bottom left) {
+      left: $arrow-offset;
+    }
+  }
+
+  &[position="top-left"] {
+    @include __body {
+      background-image: linear-gradient(
+        to bottom right,
+        transparent 50%,
+        var(--hxBackgroundColor, $gray-0) 50%
+      );
+    }
+    @include __shadow(right);
+    @include __rotateChildren(45deg, bottom right) {
+      right: $arrow-offset;
+    }
+  }
+
+  /// ======================================== ///
+  ///  RIGHT POSITIONS
+  /// ======================================== ///
+  &[position^="right"] {
+    margin-left: $arrow-margin;
+
+    @include __children {
+      left: -($arrow-margin);
+    }
+
+    @include __body {
+      background-image: linear-gradient(
+        to top left,
+        transparent 50%,
+        var(--hxBackgroundColor, $gray-0) 50%
+      );
+    }
+  }
+
+  &[position="right"], // DEPRECATED: remove in v1.0.0
+  &[position="right-middle"] {
+    @include __shadow(left);
+    @include __rotateChildren(-45deg, top left) {
+      top: 50%; // offset, tip centered
+    }
```

Comment:
### Before
In the LESS implementation, `#Arrow.children({})` and `#Arrow.rotate()` targeted the same CSS selectors (`&::before, &::after { ... }`), but were called separately, which resulted in duplicate blocks of styles with lengthy, compound selectors that were bloating the generated CSS. 

_for example:_
```css
.has-arrow[position="right"]::before, .has-arrow[position="right-middle"]::before, .has-arrow[position="right"]::after, .has-arrow[position="right-middle"]::after {
  /* children styles here */
}
.has-arrow[position="right"]::before, .has-arrow[position="right-middle"]::before, .has-arrow[position="right"]::after, .has-arrow[position="right-middle"]::after {
  /* rotate styles here */
}
```
This 162-character CSS selector was generated **twice**! (once for `#Arrow.children({})` and again for `#Arrow.rotate()`).

### Current
Now, `__rotateChildren` (initially a replacement for `#Arrow.rotate()`) accepts a content block so that we can combine both "children" and "rotate" mixins functionality to generate a _single_ block of styles for the lengthy selectors mentioned above.  This resulted in 33% less CSS being generated per Shadow DOM, thus shortening the CSS string compiled into JavaScript string, in turn reducing the file size of generated JavaScript assets.

```css
.has-arrow[position="right"]::before, .has-arrow[position="right-middle"]::before, .has-arrow[position="right"]::after, .has-arrow[position="right-middle"]::after {
  /* rotate styles here */
  /* children styles here */
}
```

This _one_ change, shaved 6KB off the size of our unminified JavaScript.

Reply:
nice! :+1:

Reply:
Great catch! 🎉 💯